Kernel/Devices/Audio/IntelHDA/Stream.h

```diff
@@ -6,26 +6,31 @@
 
 #pragma once
 
+#include <AK/Badge.h>
 #include <AK/Error.h>
 #include <AK/NonnullOwnPtr.h>
 #include <AK/OwnPtr.h>
 #include <Kernel/Devices/Audio/IntelHDA/Codec.h>
 #include <Kernel/Devices/Audio/IntelHDA/Format.h>
 #include <Kernel/Library/IOWindow.h>
 #include <Kernel/Locking/SpinlockProtected.h>
+#include <Kernel/Tasks/WaitQueue.h>
 
 namespace Kernel::Audio::IntelHDA {
 
 class Stream {
 public:
     static constexpr u32 cyclic_buffer_size_in_ms = 40;
 
+    virtual ~Stream();
+
     u8 stream_number() const { return m_stream_number; }
     bool running() const { return m_running; }
     u32 sample_rate() const { return m_format_parameters.sample_rate; }
 
-    void start();
+    virtual void start();
```

Comment:
Why does this need to be virtual? You don't override it in the OutputStream.

Reply:
It doesn't need to be, that slipped through :-) thanks!